src/realm/object-store/sync/sync_session.hpp

```diff
@@ -311,16 +314,17 @@ class SyncSession : public std::enable_shared_from_this<SyncSession> {
 
     friend class realm::SyncManager;
     // Called by SyncManager {
-    static std::shared_ptr<SyncSession> create(_impl::SyncClient& client, std::shared_ptr<DB> db, SyncConfig config,
+    static std::shared_ptr<SyncSession> create(_impl::SyncClient& client, std::shared_ptr<DB> db, RealmConfig config,
```

Comment:
Everything in the chain `SyncSession::get_session()` to `SyncSession::SyncSession()` should probably be passing `const RealmConfig&`. `RealmConfig` is sufficiently large and complicated that even moving it is not really trivial.